From a9e562b29867b5283d8225f3dc098ce3307eb191 Mon Sep 17 00:00:00 2001
From: Veerendranath Jakkam <quic_vjakkam@quicinc.com>
Date: Thu, 26 Sep 2024 15:42:37 +0530
Subject: [PATCH] Update documentation of the QCA vendor ACS channel list
 attributes

Add more detailed documentation for QCA_WLAN_VENDOR_ATTR_ACS_CH_LIST
and QCA_WLAN_VENDOR_ATTR_ACS_FREQ_LIST attributes on how the specified
channel list information is used by the driver during the ACS function.

The specified channel list represents the allowed channels for the
primary and non-primary channel operations. If any channel is not
present in the allowed channel list it shouldn't be used as a primary or
non-primary channel.

Signed-off-by: Veerendranath Jakkam <quic_vjakkam@quicinc.com>
---
 src/common/qca-vendor.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/common/qca-vendor.h b/src/common/qca-vendor.h
index c5549f6fe..78de403d7 100644
--- a/src/common/qca-vendor.h
+++ b/src/common/qca-vendor.h
@@ -2029,7 +2029,9 @@ enum qca_wlan_vendor_attr_p2p_listen_offload {
  *
  * @QCA_WLAN_VENDOR_ATTR_ACS_CH_LIST: Required and type is NLA_UNSPEC.
  * Used with command to configure channel list using an array of
- * channel numbers (u8).
+ * channel numbers (u8). This represents the list of allowed channels for
+ * the primary and non-primary channel operation. Channels which are not present
+ * in the specified list shouldn't be used as a primary or non-primary channel.
  * Note: If both the driver and user-space application supports the 6 GHz band,
  * the driver mandates use of QCA_WLAN_VENDOR_ATTR_ACS_FREQ_LIST whereas
  * QCA_WLAN_VENDOR_ATTR_ACS_CH_LIST is optional.
@@ -2061,7 +2063,10 @@ enum qca_wlan_vendor_attr_p2p_listen_offload {
  *
  * @QCA_WLAN_VENDOR_ATTR_ACS_FREQ_LIST: Required and type is NLA_UNSPEC.
  * Used with command to configure the channel list using an array of channel
- * center frequencies in MHz (u32).
+ * center frequencies in MHz (u32). This represents the list of allowed
+ * frequencies for the primary and non-primary channel operation. Frequencies
+ * which are not present in the specified list shouldn't be used as a primary or
+ * non-primary channel.
  * Note: If both the driver and user-space application supports the 6 GHz band,
  * the driver first parses the frequency list and if it fails to get a frequency
  * list, parses the channel list specified using
-- 
2.43.0

